src/components/DashBoard/DashBoard.tsx

```diff
@@ -8,171 +8,183 @@ import EventCard from "../Cards/EventCard/EventCard";
+            toast.warn("You are not signed in", { transition: Zoom });
+        }
+        console.log("BROOO",localStorage.getItem("userEmailId"))
+        const fetchData = async () => {
+            const usersRef = ref(database, "users");
+            const userEmailId = localStorage.getItem("userEmailId");
+            
 
-      if (userEmailId === null) {
-        console.error("User email ID is null");
-        return;
-      }
+            if (userEmailId === null) {
+                console.error("User email ID is null");
+                return;
+            }
 
-      const userRef = child(usersRef, userEmailId);
-      const snapshot = await get(userRef);
+            // userRef is the reference to the user's data in the database
+            const userRef = child(usersRef, userEmailId);
+            const snapshot = await get(userRef);
 
-      if (snapshot.exists()) {
-        let eventList: any[] = [];
-        if (snapshot.hasChild("createdEvents")) {
-          eventList = [
-            ...eventList,
-            ...snapshot.val().createdEvents.split(","),
-          ];
-        }
-        if (snapshot.hasChild("registeredEvents")) {
-          const registeredEvents = snapshot.val().registeredEvents.split(",");
-          registeredEvents.forEach((event: string) => {
-            if (!eventList.includes(event)) {
-              eventList.push(event);
-            }
-          });
-        }
-        
-        eventList.forEach((eventId: string) => {
-          const trimmedEventId = eventId.trim();
-          const eventsRef = ref(database, "events");
-          const eventRef = child(eventsRef, trimmedEventId);
-          get(eventRef).then((snapshot) => {
             if (snapshot.exists()) {
-              const event = snapshot.val();
-              setEventCardsData((prev: any[]) => [...prev, event]);
-              setTotalPages(Math.ceil(eventCardsData.length / itemsPerPage));
-              setIsLoading(false);
+                console.log(snapshot)
+                let eventList: any[] = [];
+                if (snapshot.hasChild("createdEvents")) {
+                    
+                    eventList = [
+                        ...eventList,
+                        ...snapshot.val().createdEvents.split(","),
+                    ];
+                }
+                if (snapshot.hasChild("registeredEvents")) {
+                    const registeredEvents = snapshot.val().registeredEvents.split(",");
+                    registeredEvents.forEach((event: string) => {
+                        if (!eventList.includes(event)) {
+                            eventList.push(event);
+                        }
+                    });
+                }
+
+                eventList.forEach((eventId: string) => {
+                    const trimmedEventId = eventId.trim();
+                    const eventsRef = ref(database, "events");
+                    
+                    if(trimmedEventId[0] =='{'){
```

Comment:
Please explain what does this part of code do.

Reply:
In the previous lines,
if (snapshot.hasChild("registeredEvents")) {
                    const registeredEvents = snapshot.val().registeredEvents.split(",");
                    registeredEvents.forEach((event: string) => {
                        if (!eventList.includes(event)) {
                            eventList.push(event);
                        }
                    });
                }
the string "{card.id}" was getting pushed into eventList and hence, giving error in the child function as '{' is not allowed.
That's why i use this to ignore this case

Reply:
This check is unnecessary, remove it.

This is the `eventList` array

![image](https://github.com/Tanay-ErrorCode/lupo-skill/assets/90561803/5d53285e-a8ce-423e-b39f-b5dfa6a764ab)